wazo_phoned/plugin_helpers/client/http.py

```diff
@@ -0,0 +1,247 @@
+            return None
+
+        next_offset = offset + limit
+        if next_offset >= results_count:
+            return None
+
+        return next_offset
+
+    def _previous_offset(self, offset, limit):
+        if offset == 0:
+            return None
+
+        if limit is None:
+            return None
+
+        previous_offset = offset - limit
+        if previous_offset < 0:
+            return 0
+
+        return previous_offset
+
+    def _get_user_tenant_uuid(self, user_uuid):
+        try:
+            return self.auth_client.users.get(user_uuid)['tenant_uuid']
+        except HTTPError as e:
+            response = getattr(e, 'response', None)
+            status_code = getattr(response, 'status_code', None)
+            if status_code == 404:
+                raise NoSuchUser(user_uuid)
+            raise
+        except ConnectionError:
+            raise WazoAuthConnectionError()
+
+
+def _build_next_url(current):
+    if current == 'menu':
+        return request.base_url.replace('menu', 'input', 1)
+    if current == 'input':
+        return request.base_url.replace('input', 'lookup', 1)
+    if current == 'lookup':
+        return request.base_url
+    return None
+
+
+class _PhoneResultFormatter:
+
+    _NAME_FIELD = 'name'
+    _NUMBER_FIELD = 'number'
+    _INVALID_CHARACTERS_REGEX = re.compile(r'[^\d*#+\(\)]+')
+    _SPECIAL_NUMBER_REGEX = re.compile(r'^\+(\d+)\(\d+\)(\d+)$')
+    _PARENTHESES_REGEX = re.compile(r'[\(\)]')
+
+    def __init__(self, lookup_results):
+        self._lookup_results = lookup_results
+        self._number_fields = self._extract_number_fields()
+        self._name_field_position = self._find_name_field_position()
+
+    def format_results(self):
+        results = []
+        for lookup_result in self._lookup_results['results']:
+            self._format_result(lookup_result, results)
+        return results
+
+    def _format_result(self, result, out):
+        for display_name, number in self._extract_result(result):
+            number = self._normalize_number(number)
+            if not number:
+                continue
+            out.append(_PhoneFormattedResult(display_name, number))
+
+    def _normalize_number(self, number):
+        number = self._extract_number_from_pretty_number(number)
+        return number
+
+    def _extract_result(self, result):
+        name = result['column_values'][self._name_field_position]
+        first_number = True
+        for candidate in self._number_fields:
+            if first_number:
+                name_result = name
+            else:
+                name_result = '{} ({})'.format(
+                    name, self._lookup_results['column_headers'][candidate].lower()
+                )
+            number = result['column_values'][candidate]
+            if not number:
+                continue
+            yield name_result, number
+            first_number = False
+
+    def _extract_number_from_pretty_number(self, pretty_number):
```

Comment:
It looks like very specific and fragile to do this
It should not be to phoned to have this logic
Moreover, it is outside of ticket

Reply:
I will add a note that it should not be in phoned but in dird.